options/options.go

```diff
@@ -75,13 +76,16 @@ func defaultSettings() *Settings {
 // The remaining options have sane defaults and are not mandatory
 func LoadFromEnvironment() error {
 	settings := defaultSettings()
-	tokeninfoURL, err := getURL("UPSTREAM_TOKENINFO_URL")
-	if err != nil {
-		return fmt.Errorf("Error with UPSTREAM_TOKENINFO_URL: %v\n", err)
+
+	if s := getString("UPSTREAM_TOKENINFO_URL", ""); s != "" {
+		tokeninfoURL, err := getURL("UPSTREAM_TOKENINFO_URL")
+		if err != nil {
+			return fmt.Errorf("Error with UPSTREAM_TOKENINFO_URL: %v\n", err)
+		}
+		settings.UpstreamTokenInfoURL = tokeninfoURL
 	}
-	settings.UpstreamTokenInfoURL = tokeninfoURL
 
-	tokeninfoURL, err = getURL("OPENID_PROVIDER_CONFIGURATION_URL")
+	tokeninfoURL, err := getURL("OPENID_PROVIDER_CONFIGURATION_URL")
```

Comment:
can you rename this variable? seems to reference tokeninfo, but its actually an openidConfigurationURL